src/Nest/QueryDsl/MultiTermQueryRewrite/RewriteMultiTerm.cs

```diff
@@ -46,10 +50,191 @@ public enum RewriteMultiTerm
+		/// </summary>
+		public static MultiTermQueryRewrite ConstantScore { get; } = new MultiTermQueryRewrite(RewriteMultiTerm.ConstantScore);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into a should clause in a boolean query, and keeps the scores
+		///  as computed by the query. Note that typically such scores are meaningless to the user, and require non-trivial
+		///  CPU to compute, so it’s almost always better to use constant_score_auto. This rewrite method will hit too many
+		///  clauses failure if it exceeds the boolean query limit (defaults to 1024).
+		/// </summary>
+		public static MultiTermQueryRewrite ScoringBoolean { get; } = new MultiTermQueryRewrite(RewriteMultiTerm.ScoringBoolean);
+
+		/// <summary>
+		/// Similar to scoring_boolean except scores are not computed. Instead, each matching document receives a constant
+		///  score equal to the query’s boost. This rewrite method will hit too many clauses failure if it exceeds the
+		/// boolean query limit (defaults to 1024).
+		/// </summary>
+		public static MultiTermQueryRewrite ConstantScoreBoolean { get; } = new MultiTermQueryRewrite(RewriteMultiTerm.ConstantScoreBoolean);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into should clause in boolean query, and keeps the scores
+		/// as computed by the query. This rewrite method only uses the top scoring terms so it will not overflow boolean
+		///  max clause count. <param name="size" /> controls the size of the top scoring terms to use.
+		/// </summary>
+		public static MultiTermQueryRewrite TopTerms(int size) => new MultiTermQueryRewrite(RewriteMultiTerm.TopTermsN, size);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into should clause in boolean query, but the scores are only
+		/// computed as the boost. This rewrite method only uses the top scoring terms so it will not overflow the boolean
+		///  max clause count. <param name="size" /> controls the size of the top scoring terms to use.
+		/// </summary>
+		public static MultiTermQueryRewrite TopTermsBoost(int size) => new MultiTermQueryRewrite(RewriteMultiTerm.TopTermsBoostN, size);
+
+		/// <summary>
+		/// A rewrite method that first translates each term into should clause in boolean query, but all term queries compute
+		///  scores as if they had the same frequency. In practice the frequency which is used is the maximum frequency of all
+		///  matching terms. This rewrite method only uses the top scoring terms so it will not overflow boolean max clause count.
+		/// <param name="size" /> controls the size of the top scoring terms to use.
+		/// </summary>
+		public static MultiTermQueryRewrite TopTermsBlendedFreqs(int size) => new MultiTermQueryRewrite(RewriteMultiTerm.TopTermsBlendedFreqsN, size);
+
+		internal static MultiTermQueryRewrite Create(string value)
+		{
+			if (value == null) throw new ArgumentNullException(nameof(value));
+
+			string rewriteType;
```

Comment:
Nit: maybe default this to `value` so we can use `var`?  Also will simplify the if statement below.